### ISSUE TITLE
Fix S3358 FP: Nested ternary operator is in a lambda

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#5773]
### Description

see https://community.sonarsource.com/t/csharpsquid-s3358-considers-lambdas-for-nesting/66955

Don't report here:
```cs
    return withTimeAccessor != null
        ? item => withTimeAccessor(item) ? ReportCellStyle.DateTime : ReportCellStyle.Date
        : item => withTime ? ReportCellStyle.DateTime : ReportCellStyle.Date;
```

### Justification

1. Different context/semantics

The lambda is a new context. A ternary is problematic when the true and/or false expressions are ternary expressions. This is not the case here. Consider e.g.

```cs
var gamble = playersAge >= 18
  ? diceRolls.Select(x => x % 2 == 0 ? "You win": "You loose")
  : Enumerable.Empty<string>();
```
 The two ternaries are running in completely different contexts.

2. Re-factoring

Nested ternaries can be refactored, e.g. like so

```cs
// Before
var greeting = isMale
  ? "Mr."
  : isMarried
    ? "Mrs."
    : "Miss";

// After
var greeting = (isMale, isMarried) switch
  {
    (true, _) => "Mr",
    (_, true) => "Mrs.",
    (_, false) => "Miss",
  };
```

I don't see a straightforward way to get rid of the nesting in the case of lambdas.
